### DISCUSSION TITLE
Is it possible to point Granian at a config file?

### DISCUSSION THREAD

**Discussion by drcongo** [emmett-framework/granian#292] [Q&A] [answered]
...as opposed to having to remember what all the command line flags are and what values they should expect. The gunicorn equivalent is to do something like `gunicorn -c /etc/gunicorn.py config.wsgi:application` which I use currently across a lot of projects. The main benefit being that I start gunicorn under supervisord, so pointing at a config file means that I can restart gunicorn with new settings without having to do `supervisorctl reread` etc. It also makes our server provisioning a bit more straight forward.

**Comment by gi0baro** [comment, marked as answer]
There's currently no support for that; but you can build an environment file using the environment variables.

**Reply by baseplate-admin** [reply]
Hey @gi0baro, are you open to add support for a config file for this?

My proposal is:
1. Read `pyproject.toml` and get config from there
2. Read 'granian` specific file?

While i agree that environment file is good enough for deployment in docker related deployment, what if i want to run granian standalone (without docker?)

**Reply by gi0baro** [reply]
@baseplate-admin I'm not a huge fan of having "deployment-related configuration" in the application code, especially given you might have multiple deployment targets for an application: using `pyproject.toml` would require you to change the application code every time you want to change the deployment configuration and would also make impossible to have different configurations.

For deployments not using docker/kubernetes I expect you to use some supervisor like SystemD, thus having a service unit still support environment files.
I don't really see the point in defining _another configuration standard_ for this, probably it's faster (and simpler) for the specific deployment scenario to wrap granian launch into a bash script, or launch granian using its Python interface: at that point you can implement whatever configuration parsing you need/prefer.
